internal/controller/stages/argocd.go

```diff
@@ -88,15 +89,49 @@ func (r *reconciler) checkHealth(
 					}
 				}
 			}
-			if healthy, reason := libArgoCD.IsApplicationHealthyAndSynced(
-				app,
-				desiredRevision,
-			); !healthy {
-				health.Status = kargoapi.HealthStateUnhealthy
-				health.Issues = append(health.Issues, reason)
-			}
+			health = health.Merge(stageHealthForAppHealth(app))
+			health = health.Merge(stageHealthForAppSync(app, desiredRevision))
 		}
 	}
 
 	return health
 }
+
+func stageHealthForAppHealth(app *argocd.Application) kargoapi.Health {
+	switch app.Status.Health.Status {
+	case argohealth.HealthStatusDegraded, argohealth.HealthStatusUnknown:
+		return kargoapi.Health{
+			Status: kargoapi.HealthStateUnhealthy,
+			Issues: []string{fmt.Sprintf("Argo CD Application %q in namespace %q has health state %q",
+				app.Name, app.Namespace, app.Status.Health.Status)},
+		}
+	case argohealth.HealthStatusProgressing, "":
+		return kargoapi.Health{
+			Status: kargoapi.HealthStateProgressing,
+			Issues: []string{fmt.Sprintf("Argo CD Application %q in namespace %q is progressing",
+				app.Name, app.Namespace)},
+		}
+	default:
+		return kargoapi.Health{Status: kargoapi.HealthStateHealthy}
+	}
+}
+
+func stageHealthForAppSync(app *argocd.Application, revision string) kargoapi.Health {
+	if app.Status.Sync.Status != argocd.SyncStatusCodeSynced ||
```

Comment:
I think if we change this line a little, we can fix #670 and close #675 without merging.

Reply:
Although, [this](https://github.com/akuity/kargo/pull/675#issuecomment-1711021158) would also have to be addressed.

Reply:
Nevermind. Will just rebase #670 on this.